shell/platform/android/test/io/flutter/embedding/engine/systemchannels/PlatformChannelTest.java

```diff
@@ -113,6 +115,61 @@ public void setSystemExclusionRectsSendsJSONExceptionOnIncorrectDataShape() thro
         );
     }
 
+    @Test
+    public void getSystemExclusionRectsSendsExclusionRectsToFrameworkOnSuccess() throws JSONException {
+        DartExecutor dartExecutor = mock(DartExecutor.class);
+        PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
+        PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
+        platformChannel.setPlatformMessageHandler(platformMessageHandler);
+
+        int top = 0;
+        int right = 500;
+        int bottom = 250;
+        int left = 0;
+
+        ArrayList<Rect> expectedExclusionRects = new ArrayList<Rect>();
+        Rect gestureRect = new Rect(left, top, right, bottom);
+        expectedExclusionRects.add(gestureRect);
+        when(platformMessageHandler.getSystemGestureExclusionRects()).thenReturn(expectedExclusionRects);
+
+        MethodCall callGetSystemGestureExclusionRects = new MethodCall(
+            "SystemGestures.getSystemGestureExclusionRects",
+            null
+        );
+        ResultsMock resultsMock = mock(ResultsMock.class);
+        platformChannel.parsingMethodCallHandler.onMethodCall(callGetSystemGestureExclusionRects, resultsMock);
+
+        ArrayList<HashMap<String, Integer>> expectedEncodedOutputRects = new ArrayList<HashMap<String, Integer>>();
+        HashMap<String, Integer> rectMap = new HashMap<String, Integer>();
+        rectMap.put("top", top);
```

Comment:
In tests it is often preferable to avoid using variables for verification purposes. Developers will need to read every line of the test to understand what the expected values are. It essentially moves logic into the test, when a test should only include setup and expectations (or as close to that as possible). I would recommend putting the literal values in this map, rather than the variables.

Reply:
Sounds good. I was thinking it would be more clear to have both values created from the same variable so that developers know that the API output values should match the parsed values